dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java

```diff
@@ -318,6 +321,8 @@ private InputStream buildVelocity(final Container container, final String uuid,
                     velocityCodeBuilder.append("<div")
                         .append(" data-dot-object=")
                         .append("\"contentlet\"")
+                        .append(" data-dot-on-number-of-pages=")
+                        .append("\"$ON_NUMBER_OF_PAGES\"")
```

Comment:
are you sure that you need both?, I mean are you sure that you need the velocity variable and the data attribute too?

Reply:
yes we need it. Please check the related ticket https://github.com/dotCMS/core/issues/24217 thanks